From 0b9e087f8f04a106916f03d01acee3ab5d09d144 Mon Sep 17 00:00:00 2001
From: Quanyang Wang <quanyang.wang@windriver.com>
Date: Tue, 10 Jan 2023 10:50:03 +0800
Subject: [PATCH] s32: serdes: hwconfig: change hwconfig for CVC board.

At CVC board, the serdes lanes configuration is as below:

Serdes0_lane0: Pcie
Serdes0_lane1: PFE2_SGMII
Serdes1_lane0: PFE0_SGMII
Serdes1_lane1: PFE1_SGMII

Upstream-Status: Inappropriate [WR Linux specific change]

Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
[fix: add S32CC_HWCONFIG in board/nxp/aptiv_cvc_fl/Kconfig]
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 board/nxp/aptiv_cvc_fl/Kconfig | 23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)

diff --git a/board/nxp/aptiv_cvc_fl/Kconfig b/board/nxp/aptiv_cvc_fl/Kconfig
index b8d38ddd9a..9165986ae2 100644
--- a/board/nxp/aptiv_cvc_fl/Kconfig
+++ b/board/nxp/aptiv_cvc_fl/Kconfig
@@ -12,4 +12,27 @@ config SYS_BOARD
 config NR_DRAM_BANKS
 	default 2
 
+config S32CC_HWCONFIG
+        string "S32CC HWConfig definition"
+        depends on PHY_S32CC_SERDES
+	default "serdes0:mode=pcie&xpcs1,clock=int,fmhz=100;pcie0:mode=rc;xpcs0_1:speed=1G;serdes1:mode=xpcs0&xpcs1,clock=int,fmhz=100;xpcs1_0:speed=1G;xpcs1_1:speed=1G"
+        help
+          The configuration for the SerDes modules, stored in
+          the variable 'hwconfig'.
+          It configures generic attributes for 'serdesX', such as 'mode' (see below),
+          the clock type (internal or external),
+          clock frequency 'fmhz' (100 or 125, SGMII only modes), 'pcieX' mode
+          ('rc' or 'ep') and the 'xpcsX_Y' interfaces for any of the two lanes.
+          In the atributes presented above and below, 'X' (0 or 1) is the index
+          of the SerDes module and 'Y' (0 or 1) is the index of the XPCS interface.
+          SerDes 'mode' attribute can be 'pcie' (PCIe only mode), 'pcie&xpcsY' (PCIe
+          and SGMII combo mode), 'xpcs0&xpcs1' (SGMII only mode).
+          For the combo modes 'pcie&xpcsY', lane0 is PCIe and lane1
+          is connected to a GMAC or PFE MAC depending on 'Y' and 'fmhz' (which
+          match a certain SerDes configuration from the SerDes manual).
+          PCIe always works at 100MHz, therefore 'fmhz' is only for SGMII only mode.
+          Config 'xpcsX_Y' is used to specifically configure each of the two SGMII
+          PHYs, by setting the 'speed' (10M, 100M, 1G and 2G5).
+          S32G3 has 1 GMAC running at 1Gbps and 3 PFE MACs, running at 10Mbps - 2.5Gbps speeds.
+
 endif
-- 
2.34.1

